pkg/client/perseshttp/client.go

```diff
@@ -102,7 +103,23 @@ type RESTClient struct {
 	// base is the root URL for all invocations of the client
 	BaseURL *url.URL
 	// Set specific behavior of the client.  If not set http.DefaultClient will be used.
-	Client *http.Client
+	Client    *http.Client
+	authMutex sync.RWMutex
+}
+
+func (c *RESTClient) SetBasicAuth(username, password string) {
```

Comment:
What is the difference from having it configured in the rest config client ? I mean for the basic Auth in particular 
For authorization I would maybe understand even if it is for client credentials, I'd prefer to have a way to configure my own client coming from oAuth2 library like I did in my Pr

Reply:
it will avoid recreating entirely the RESTClient.